### PR TITLE
[Impeller] emulate gl blitframebuffer for GLES 2.0

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/157064
Fixes https://github.com/flutter/flutter/issues/135764

Emulates gl.BlitFramebuffer with a draw call. manual testing with gl.BlitFramebuffer disabled works for me locally...

### CODE REVIEW

File: impeller/renderer/backend/gles/blit_command_gles.cc [hunk 1]
```diff
@@ -66,16 +66,78 @@ std::string BlitCopyTextureToTextureCommandGLES::GetLabel() const {
   return label;
 }
 
+namespace {
+static GLfloat pixels_to_gl_coords(GLfloat position, GLfloat pixels) {
```
Comment (gaaclarke) [inline impeller/renderer/backend/gles/blit_command_gles.cc:70]:
```suggestion
/// Maps `position` in non-normalized texture pixels in the range [0,`pixels`] to a square at (x:-1, y:-1, width:2, height:2).
static GLfloat pixels_to_gl_coords(GLfloat position, GLfloat pixels) {
```

File: impeller/renderer/backend/gles/blit_command_gles.cc [hunk 2]
```diff
@@ -66,16 +66,78 @@ std::string BlitCopyTextureToTextureCommandGLES::GetLabel() const {
...
     const ReactorGLES& reactor) const {
   const auto& gl = reactor.GetProcTable();
 
   // glBlitFramebuffer is a GLES3 proc. Since we target GLES2, we need to
   // emulate the blit when it's not available in the driver.
   if (!gl.BlitFramebuffer.IsAvailable()) {
-    // TODO(157064): Emulate the blit using a raster draw call here.
-    VALIDATION_LOG << "Texture blit fallback not implemented yet for GLES2.";
-    return false;
+    if (!blit_program.has_value()) {
```
Comment (gaaclarke) [inline impeller/renderer/backend/gles/blit_command_gles.cc:82]: This whole branch should be its own function.

File: impeller/renderer/backend/gles/blit_command_gles.cc [hunk 3]
```diff
@@ -66,16 +66,78 @@ std::string BlitCopyTextureToTextureCommandGLES::GetLabel() const {
...
+    draw_fbo = dst_handle.value();
+
+    gl.Disable(GL_BLEND);
+    gl.Disable(GL_SCISSOR_TEST);
+    gl.Disable(GL_DITHER);
+    gl.Disable(GL_CULL_FACE);
+    gl.ColorMask(GL_TRUE, GL_TRUE, GL_TRUE, GL_TRUE);
+
+    gl.ClearColor(0.0f, 0.0f, 0.0f, 1.0f);
+    gl.Clear(GL_COLOR_BUFFER_BIT);
```
Comment (gaaclarke) [inline impeller/renderer/backend/gles/blit_command_gles.cc:109]: Do we need to clear, aren't we drawing fully to the FBO?

File: impeller/renderer/backend/gles/blit_command_gles.cc (hunk 1 above)
Comment (gaaclarke) [inline impeller/renderer/backend/gles/blit_command_gles.cc:70]: This doesn't match the style guides naming structure: 
https://google.github.io/styleguide/cppguide.html#Function_Names

File: impeller/renderer/backend/gles/blit_command_gles.cc [hunk 4]
```diff
@@ -66,16 +66,78 @@ std::string BlitCopyTextureToTextureCommandGLES::GetLabel() const {
...
+    GLint position_index = 0;
+    gl.EnableVertexAttribArray(position_index);
+    gl.VertexAttribPointer(position_index, 2, GL_FLOAT, GL_FALSE,
+                           sizeof(GLfloat) * 4, (void*)0);
+    GLint texcoord_index = 1;
+    gl.EnableVertexAttribArray(texcoord_index);
+    gl.VertexAttribPointer(texcoord_index, 2, GL_FLOAT, GL_FALSE,
+                           sizeof(GLfloat) * 4, (void*)(sizeof(GLfloat) * 2));
+
+    gl.DrawArrays(GL_TRIANGLES, 0, 6);
```
Comment (gaaclarke) [inline impeller/renderer/backend/gles/blit_command_gles.cc:138]: Wouldn't a triangle strip be more efficient since it only needs 4 indices.

Comment (jonahwilliams) [reply impeller/renderer/backend/gles/blit_command_gles.cc]: Done

File: impeller/renderer/backend/gles/blit_command_gles.cc (hunk 1 above)
Comment (jonahwilliams) [reply impeller/renderer/backend/gles/blit_command_gles.cc]: Done

File: impeller/renderer/backend/gles/blit_command_gles.cc (hunk 2 above)
Comment (jonahwilliams) [reply impeller/renderer/backend/gles/blit_command_gles.cc:181]: Done

File: impeller/renderer/backend/gles/blit_command_gles.cc (hunk 1 above)
Comment (jonahwilliams) [reply impeller/renderer/backend/gles/blit_command_gles.cc]: Done

**Comment by jonahwilliams** [comment]
Still working on this

**Pull request closed without merging**